infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -171,33 +174,40 @@ public SequencerServer(ServerContext serverContext) {
                 && Long.parseLong((String) opts.get("--sequencer-cache-size")) != 0) {
             cacheSize = Long.parseLong((String) opts.get("--sequencer-cache-size"));
         }
+        if (opts.get("--sequencer-per-stream-cache-size") != null
+                && Long.parseLong((String) opts.get("--sequencer-per-stream-cache-size")) != 0) {
+            perStreamCacheSize =
+                    Long.parseLong((String) opts.get("--sequencer-per-stream-cache-size"));
+        }
+
 
         conflictToGlobalTailCache = Caffeine.newBuilder()
```

Comment:
Do you anticipate the streams to be in such large number that you need a loading cache for them ?